### DISCUSSION TITLE
How to draw the cropped bounding box after cropping?

### DISCUSSION THREAD

**Discussion by AlbertLiu1974** [CropperBlazor/Cropper.Blazor#322] [General]
Hi All, 

  I want to record the bounding box and draw a red rectangle to save it for review purpose. Kindly refer to the image. 

![image](https://github.com/CropperBlazor/Cropper.Blazor/assets/152601781/4bfe31db-d939-4ddd-919b-a8d4b6e56d1c)

Not sure how to do it in blazor. Kindly help. I am using .net 8.1 

Br, Albert

**Comment by MaxymGorn** [comment]
Hi @AlbertLiu1974. Unfortunately, Cropper.Blazor  library don't support drawing functionality.

But, If you want to draw images in Blazor, you should to use canvas javascript functionality for that. You can try to use ready library for working with canvas in blazor: https://github.com/BlazorExtensions/Canvas. Definitely, you can another more advanced libraries if it requires or call drawing functions from canvas object reference.

You can get canvas element from cropped image using following example: https://cropperblazor.github.io/examples/cropping#advanced-usage

Example code:
```
// Get a reference to a JavaScript cropped canvas object.
        CroppedCanvas croppedCanvas = await cropperComponent!.GetCroppedCanvasAsync(getCroppedCanvasOptions);
        // Get reference canvas object when you have been cropping image.
        croppedCanvas!.JSRuntimeObjectRef
```

**Reply by AlbertLiu1974** [reply]
Hi @MaxymGorn  Yes. I can crop the image so far and thanks for the CroppedCanvas and also I can get the bounding box coordinate as well from the OnCropEvent handle.  Now It seems I can't combine the BlazorCanvas working with CropperComponent.  I. only can create a separate page and summarize the cropped images in this page via Canvas.  Or there is another way? 

Best regards,
Albert Liu

**Reply by MaxymGorn** [reply]
Hi @AlbertLiu1974. Cropper js can work via canvas instead of img, but it requires small changes in library, because Cropper.Blazor library support only img rendering. I will roll out fix for that.

**Reply by AlbertLiu1974** [reply]
Hi @MaxymGorn , Thank you for the help. I think if there has a canvas back from Cropper.Blazor, will solve two my issues, 1. for drawing shapes.  2. for cropping via polygon(around the object)--> Hope I am right. Thanks a lot. 

simple demo: https://youtu.be/ZMA7CQGMJAk

Best regards,
Albert Liu
